zebra/zebra_fpm_netlink.c

```diff
@@ -189,7 +192,12 @@ static int netlink_route_info_add_nh(struct netlink_route_info *ri,
 
 	if (nexthop->type == NEXTHOP_TYPE_IPV6
 	    || nexthop->type == NEXTHOP_TYPE_IPV6_IFINDEX) {
-		nhi.gateway = &nexthop->gate;
+		/* Special handling for IPv4 route with IPv6 Link Local next hop
```

Comment:
Won'tthis break v4 routes with v4 nexthops?

Reply:
As I understand the particular condition will not hit for v4  routes with v4 nexthops, so those should not be impacted.